centreon/www/front_src/src/Header/specs/Header.Hosts.ts

```diff
@@ -0,0 +1,258 @@
+          pending: '0'
+        };
+
+        initialize({ hosts_status: hoststubs });
+        getElements();
+
+        cy.get('@serviceButton').within(() => {
+          cy.get('.MuiBadge-badge.MuiBadge-colorPending')
+            .should('exist')
+            .should('not.be.visible');
+        });
+      });
+    });
+
+    describe('top status counter', () => {
+      it('should display status counter numbers with the desired format', () => {
+        // given
+        const hoststubs = {
+          down: { unhandled: '12' },
+          ok: '12134',
+          unreachable: { unhandled: '126' }
+        };
+
+        initialize({ hosts_status: hoststubs });
+        getElements();
+
+        // 12 => 12
+        cy.get('@downCounter').should('be.visible').contains('12');
+
+        // 125 => 125
+        cy.get('@unreachableCounter').should('be.visible').contains('126');
+
+        // 12134 => 12.1k
+        cy.get('@upCounter').should('be.visible').contains('12.1k');
+
+        cy.matchImageSnapshot();
+      });
+
+      it('should redirect on click on the counter', () => {
+        // given
+        const hoststubs = {
+          critical: { unhandled: '12' },
+          ok: '12134',
+          unknown: { unhandled: '125' },
+          warning: { unhandled: '14688222' }
+        };
+
+        initialize({ hosts_status: hoststubs });
+        getElements();
+
+        cy.get('@downCounter').click();
+
+        cy.url().should(
+          'include',
+          'monitoring/resources?filter={%22criterias%22:[{%22name%22:%22resource_types%22,%22value%22:[{%22id%22:%22host%22,%22name%22:%22Host%22}]},{%22name%22:%22statuses%22,%22value%22:[{%22id%22:%22DOWN%22,%22name%22:%22Down%22}]},{%22name%22:%22states%22,%22value%22:[{%22id%22:%22unhandled_problems%22,%22name%22:%22Unhandled%22}]},{%22name%22:%22search%22,%22value%22:%22%22}]}&fromTopCounter=true'
+        );
+
+        cy.get('@unreachableCounter').click();
+
+        cy.url().should(
+          'include',
+          'monitoring/resources?filter={%22criterias%22:[{%22name%22:%22resource_types%22,%22value%22:[{%22id%22:%22host%22,%22name%22:%22Host%22}]},{%22name%22:%22statuses%22,%22value%22:[{%22id%22:%22UNREACHABLE%22,%22name%22:%22Unreachable%22}]},{%22name%22:%22states%22,%22value%22:[{%22id%22:%22unhandled_problems%22,%22name%22:%22Unhandled%22}]},{%22name%22:%22search%22,%22value%22:%22%22}]}&fromTopCounter=true'
+        );
+
+        cy.get('@upCounter').click();
+
+        cy.url().should(
+          'include',
+          'monitoring/resources?filter={%22criterias%22:[{%22name%22:%22resource_types%22,%22value%22:[{%22id%22:%22host%22,%22name%22:%22Host%22}]},{%22name%22:%22statuses%22,%22value%22:[{%22id%22:%22UP%22,%22name%22:%22Up%22}]},{%22name%22:%22states%22,%22value%22:[]},{%22name%22:%22search%22,%22value%22:%22%22}]}&fromTopCounter=true'
+        );
+      });
+    });
+
+    describe('sub menu', () => {
+      it('should have a button to open the submenu', () => {
```

Comment:
```suggestion
      it('displays the submenu when the toggle icon is clicked', () => {
```

Reply:
well it's not exactly what happens, since now, the button is englobing the header icon, the text and the toggle icon.... 
so `should open the submenu when clicking on the button` seems more appropriate

Reply:
Ok but don't use "should". Because it means the test might fail sometimes. Starting by an imperative verb means the test have to pass